spec/extensions/openapi_spec.cr

```diff
@@ -0,0 +1,173 @@
+      "cast"
+    ],
+    "type": "object",
+    "properties": {
+      "string": {
+        "type": "string"
+      },
+      "opt_string": {
+        "type": "string",
+        "readOnly": true
+      },
+      "inner_schema": {
+        "$ref": "#/components/schemas/Model_InnerModel"
+      },
+      "cast": {
+        "type": "string",
+        "example": "1"
+      }
+    }
+  }
+  JSON
+
+  struct InnerModel
+    extend OpenAPI::Generator::Serializable
+    include JSON::Serializable
+
+    @[OpenAPI::Field(write_only: true)]
+    property array_of_int : Array(Int32)
+
+    SCHEMA = <<-JSON
+    {
+      "required": [
+        "array_of_int"
+      ],
+      "type": "object",
+      "properties": {
+        "array_of_int": {
+          "type": "array",
+          "items": {
+            "type": "integer"
+          },
+          "writeOnly": true
+        }
+      }
+    }
+    JSON
+  end
+
+  struct ComplexModel
+    extend OpenAPI::Generator::Serializable
+    include JSON::Serializable
+
+    property union_types : Int32 | String | Hash(String, InnerModel)
+    property free_form : JSON::Any
+    property array_of_hash : Array(Hash(String, Int32 | String))
+
+    SCHEMA = <<-JSON
+    {
+      "required": [
+        "union_types",
+        "free_form",
+        "array_of_hash"
+      ],
+      "type": "object",
+      "properties": {
+        "union_types": {
+          "oneOf": [
+            {
+              "type": "object",
+              "additionalProperties": {
+                "$ref": "#/components/schemas/Model_InnerModel"
+              }
+            },
+            {
+              "type": "integer"
+            },
+            {
+              "type": "string"
+            }
+          ]
+        },
+        "free_form": {
+          "type": "object",
+          "additionalProperties": true
+        },
+        "array_of_hash": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "additionalProperties": {
+              "oneOf": [
+                {
+                  "type": "integer"
+                },
+                {
+                  "type": "string"
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+    JSON
+  end
+end
+
+class ClearModelExample
+  include Clear::Model
+  extend OpenAPI::Generator::Serializable
+
+  column id : Int64, primary: true, api_read_only: true, example: "123"
+  column email : String, api_write_only: true, example: "default@gmail.com"
+end
+
+struct ClearModelExampleCopy
+  extend OpenAPI::Generator::Serializable
+  include JSON::Serializable
+
+  @[OpenAPI::Field(read_only: true, example: "123")]
+  property id : Int64
+
+  @[OpenAPI::Field(write_only: true, example: "default@gmail.com")]
+  property email : String
+end
+
+describe OpenAPI::Generator::Serializable do
+  it "should serialize an object into an openapi schema" do
+    json_schema = ::Model.generate_schema.to_pretty_json
```

Comment:
Why are you testing object serialisation here? Isn't this tested in OpenAPI generator already?

Reply:
I want to make sure this branch is running

https://github.com/place-labs/clear/blob/2d25af8075e765e34c1b9bd29f6b8956acbe0c9f/src/clear/extensions/openapi.cr#L36